### PR TITLE
Add Prop 225 to v17 Upgrade Handler

### PR DESCRIPTION
**Pull request by shellvish**
This PR adds the execution of Prop 225 to the v17 Upgrade Handler.

Relevant details can be found [here](https://www.mintscan.io/stride/proposals/225) and [here](https://commonwealth.im/stride/discussion/14238-shd-strd-protocol-liquidity-matching-pt2). 

In particular, 31,572.30 STRD should be released to stride1auhjs4zgp3ahvrpkspf088r2psz7wpyrypcnal from [Community Pool Growth](https://www.mintscan.io/stride/address/stride1lj0m72d70qerts9ksrsphy9nmsd4h0s88ll9gfphmhemh8ewet5qj44jc9).

### CODE REVIEW

**Comment by asalzmann** [comment]
adding here makes sense, just checking - after the upgrade, this would also be possible through an executable gov prop, right?

**Comment by shellvish** [comment]
> adding here makes sense, just checking - after the upgrade, this would also be possible through an executable gov prop, right?

I believe that's right!

File: app/upgrades.go [hunk 1]
```diff
@@ -227,6 +227,7 @@ func (app *StrideApp) setupUpgradeHandlers(appOpts servertypes.AppOptions) {
 			app.InterchainqueryKeeper,
 			app.RatelimitKeeper,
 			app.StakeibcKeeper,
+			app.BankKeeper,
```
Comment (sampocs) [inline app/upgrades.go:230]: the absolute pico-est nit of all time, but I usually like to keep these keepers sorted in alphabetical order which makes it easier to line up the function signature whenever this list changes!

File: app/upgrades/v17/upgrades.go [hunk 2]
```diff
@@ -60,6 +62,12 @@ var (
...
 	OsmosisTransferChannelId = "channel-5"
+
+	// Constants for Prop 225
+	CommunityPoolGrowthAddress = "stride1lj0m72d70qerts9ksrsphy9nmsd4h0s88ll9gfphmhemh8ewet5qj44jc9"
+	LiquidityReceiver          = "stride1auhjs4zgp3ahvrpkspf088r2psz7wpyrypcnal"
+	Prop225TransferAmount      = sdk.NewInt(31_572_300_000)
+	Ustrd                      = "ustrd"
 )
 
 // CreateUpgradeHandler creates an SDK upgrade handler for v15
```
Comment (riley-stride) [inline app/upgrades/v17/upgrades.go:73]: nit: 
```suggestion
// CreateUpgradeHandler creates an SDK upgrade handler for v17
```

File: app/upgrades/v17/upgrades.go [hunk 3]
```diff
@@ -298,3 +312,11 @@ func AddRateLimitToOsmosis(ctx sdk.Context, k ratelimitkeeper.Keeper) error {
 
 	return nil
 }
+
+// Execute Prop 225, release STRD to stride1auhjs4zgp3ahvrpkspf088r2psz7wpyrypcnal
+func ExecuteProp225(ctx sdk.Context, k bankkeeper.Keeper) error {
+	communityPoolGrowthAddress := sdk.MustAccAddressFromBech32(CommunityPoolGrowthAddress)
+	liquidityReceivernAddress := sdk.MustAccAddressFromBech32(LiquidityReceiver)
```
Comment (riley-stride) [inline app/upgrades/v17/upgrades.go:319]: nit: 
```suggestion
	liquidityReceiverAddress := sdk.MustAccAddressFromBech32(LiquidityReceiver)
```

File: app/upgrades/v17/upgrades.go [hunk 4]
```diff
@@ -298,3 +312,11 @@ func AddRateLimitToOsmosis(ctx sdk.Context, k ratelimitkeeper.Keeper) error {
...
+	transferCoin := sdk.NewCoin(Ustrd, Prop225TransferAmount)
+	return k.SendCoins(ctx, communityPoolGrowthAddress, liquidityReceivernAddress, sdk.NewCoins(transferCoin))
```
Comment (riley-stride) [inline app/upgrades/v17/upgrades.go:321]: nit: 
```suggestion
	return k.SendCoins(ctx, communityPoolGrowthAddress, liquidityReceiverAddress, sdk.NewCoins(transferCoin))
```

**Review by riley-stride — APPROVED** [review]

**Review by sampocs — APPROVED** [review]
lgtm, thanks for taking this one! 

I verified the addresses match the forum post - is there an additional step I should do to verify either of these?

File: app/upgrades/v17/upgrades.go (hunk 2 above)
Comment (sampocs) [reply app/upgrades/v17/upgrades.go:73]: just update this in the other PR!

**Pull request merged**